### DISCUSSION TITLE
visidata way of doing groupby.

### DISCUSSION THREAD

**Discussion by Awannaphasch2016** [saulpw/visidata#2463] [Q&A]
says I have col1, col2, col3 as columns

I want to group using col1.
col2 determine rank in col1.
find differences between col3 value that has first rank (aka min(col2)) and col3 value that has last rank (aka max(col3))

is there a visidata way to go about it? I don't want to leave visidata. haha

**Comment by Awannaphasch2016** [comment]
@dngor (handle in discord) answer my question in visidata discord channel. I recite his respond here to help other who encounter similar problem in the future.

his first attempt fails. @saulpw states that this might be an issue of aggregators, that might get addressed as part of  #2380 
"""
1. make col1 and col2 key columns (!)
2. g[ to sort the table 
3. 3. remove col2 as a key column (!)
4. 4. aggregate col3 as a list (+list)
5. F on col1 (the resulting col3 list is grouped on col1 and in test is ordered by col2, but maybe this is just coincidence?)
6. =col3_list[-1] - col3_list[0] (This part fails bcause col3_list contains asn error because +list aggregator is producing an error "int argument must be a string, a bytes-like object, or a real number, not 'list')
"""

he found a solution in this next attempt
"""
Looks like col3's type is propagating to col3_list. col3_list doesn't acquire an error if col3 is untyped. 
You can then =col3_list[0] and =col3_list[-1] to extract the first and last value, rename the columns, set them to a numeric type, and then =last-first to get an answer
"""

**Reply by reagle** [reply]
(Sounds like this was answered then though it still shows as unanswered.)

**Comment by midichef** [comment]
Also, if you want more groupby featuers, we are drafting a feature to add a column of rankings/sums/averages/etc when grouping on key columns. You can check out the discussion:
https://github.com/saulpw/visidata/pull/2417
